protected/humhub/modules/user/controllers/AuthController.php

```diff
@@ -223,20 +252,11 @@ protected function login($user, $authClient)
         $redirectUrl = ['/user/auth/login'];
         $success = false;
         $this->trigger(static::EVENT_BEFORE_CHECKING_USER_STATUS, new UserEvent(['user' => $user]));
-        if ($user->status == User::STATUS_ENABLED) {
-            $duration = 0;
-            if (
-                ($authClient instanceof BaseFormAuth && $authClient->login->rememberMe) ||
-                !empty(Yii::$app->session->get('loginRememberMe'))) {
-
-                $duration = Yii::$app->getModule('user')->loginRememberMeDuration;
-            }
-            (new AuthClientService($authClient))->updateUser($user);
 
-            if ($success = Yii::$app->user->login($user, $duration)) {
-                Yii::$app->user->setCurrentAuthClient($authClient);
-                $redirectUrl = Yii::$app->user->returnUrl;
-            }
+        if ($user->status == User::STATUS_ENABLED) {
+            [$success, $redirectUrl] = $this->doLogin($user, $authClient, $redirectUrl);
+        } elseif ($user->status == User::STATUS_SOFT_DELETED && $authClient->getId() !== 'local') {
```

Comment:
@yuriimaz Since it is not possible for SoftDeleted users to be reactivated, we can remove this as well, can't we?

Anyway I would keep the  `doLogin` method, because this way the code is a bit cleaner.

Reply:
Ok, removed